src/main/java/bio/terra/cli/service/WorkspaceManagerService.java

```diff
@@ -242,17 +243,23 @@ public WorkspaceDescription createWorkspace(
                   CREATE_WORKSPACE_MAXIMUM_RETRIES,
                   CREATE_WORKSPACE_DURATION_SLEEP_FOR_RETRY);
           logger.debug("create workspace context result: {}", createContextResult);
-
-          // if this is a spend profile access denied error, then throw a more user-friendly error
-          // message
-          if (createContextResult.getJobReport().getStatus().equals(JobReport.StatusEnum.FAILED)
-              && createContextResult.getErrorReport().getMessage().contains("spend profile")
-              && createContextResult.getErrorReport().getStatusCode()
-                  == HttpStatusCodes.STATUS_CODE_FORBIDDEN) {
-            throw new UserActionableException(
-                "Accessing the spend profile failed. Ask an administrator to grant you access.");
+          StatusEnum status = createContextResult.getJobReport().getStatus();
+          if (StatusEnum.FAILED == status) {
+            // need to delete the empty workspace before continuing
+            HttpUtils.callWithRetries(
+                () -> workspaceApi.deleteWorkspace(workspaceId),
+                WorkspaceManagerService::isRetryable);
```

Comment:
I think we should handle failures in this delete call a little differently. I'm thinking that if this workspace delete failed, but the original error was no spend profile access, then it would be confusing to get a "failed to delete workspace" error.

Not sure the best way to implement. Maybe just a `logger.error` for errors here and then fall through to throwing the original exception below?

Reply:
Are you thinking we need to put a try/catch around the delete call and swallow the exception and just log it? That would be a bit tricky to read.

Reply:
I don't think it matters so much which error/exception you throw -- the one from creation or deletion. As long as both are logged we should be able to debug.

It's more that a custom error message would be helpful e.g. "Workspace creation failed and there was a problem cleaning up. You may see a partially created workspace in your list of available ones.". Otherwise, I'm thinking we'll get reports like "I tried to create a workspace, but got a message about delete failing.", which would be confusing to me as a user.

So yes one approach to doing that could be to `try/catch` around this delete, wrap that exception in a custom error message and then re-throw. Or you could `try/catch` around this delete, log that exception, wrap the original create exception in a custom error message and throw that. I was thinking the latter is slightly better because you're not double-wrapping -- i.e. the `deleteWorkspace` call already wraps in a CLI exception I think.

Reply:
I think I have a solution with two error messages, one for the spend profile, and another with that message and some text about the deletion failure.